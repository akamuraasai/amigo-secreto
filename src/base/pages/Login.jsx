import React, { Component } from 'react';
import { Grid, Header, Segment, Message, Form, Button } from 'semantic-ui-react';

class Login extends Component {
  constructor() {
    super();

    this.state = { email: '', pass: '' };
  }

  setText = field => evt => {
    this.setState({ [field]: evt.target.value });
  }

  render () {
    return (
      <div className='login-form'>
        <Grid textAlign='center' style={{ height: '100%' }} verticalAlign='middle'>
          <Grid.Column style={{ maxWidth: 450 }}>
            <Header as='h4' textAlign='center'>
              <Header as='h1' color='orange' style={{ marginBottom: 16 }}>Amigo Secreto</Header>
              &nbsp;Acesse sua conta
            </Header>
            <Form size='large'>
              <Segment stacked>
                <Form.Input
                  fluid
                  icon='user'
                  iconPosition='left'
                  placeholder='E-mail'
                  id='email'
                  onChange={this.setText('email')}
                />
                <Form.Input
                  fluid
                  icon='lock'
                  iconPosition='left'
                  placeholder='Senha'
                  type='password'
                  id='pass'
                  onChange={this.setText('pass')}
                />
                <Button color='orange' fluid size='large'>Entrar</Button>
              </Segment>
            </Form>
            <Message>
              Novo usu√°rio? <a href='#'>Cadastre-se</a>
            </Message>
          </Grid.Column>
        </Grid>
      </div>
    );
  }
}

export default Login;
