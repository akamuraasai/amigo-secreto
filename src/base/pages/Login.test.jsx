import React from 'react';
import { shallow } from 'enzyme';

import Login from './Login';

describe('Pagina de Login', () => {
  const login = shallow(<Login />);

  it('deve renderizar sem problemas', () => {
    expect(login).toMatchSnapshot();
  });

  describe('Formul치rio de Login', () => {
    it('tem de haver um formul치rio na pagina', () => {
      expect(login.find('Form').exists()).toBe(true);
    });

    it('tem que ter um input para o e-mail de login', () => {
      expect(login.find('#email').exists()).toBe(true);
    });

    it('tem que ter um input para a senha de login', () => {
      expect(login.find('#pass').exists()).toBe(true);
    });

    describe('Quando o usu치rio digita no input de e-mail', () => {
      const email = 'john@mail.com';

      beforeEach(() => {
        login.find('#email').simulate('change', { target: { value: email } });
      });

      it('atualiza o valor do state inserindo o que foi digitado no input', () => {
        expect(login.state().email).toBe(email);
      });
    });

    describe('Quando o usu치rio digita no input de senha', () => {
      const pass = '123456';

      beforeEach(() => {
        login.find('#pass').simulate('change', { target: { value: pass } });
      });

      it('atualiza o valor do state inserindo o que foi digitado no input', () => {
        expect(login.state().pass).toBe(pass);
      });
    });
  });

  describe('Mensagem de Novo Cadastro', () => {
    it('tem que ter uma mensagem com link para novos cadastros', () => {
      expect(login.find('Message').exists()).toBe(true);
    });

    it('tem que ter o link para a rota de novos cadastros', () => {
      expect(login.find('Message > a').exists()).toBe(true);
    });
  });
});
