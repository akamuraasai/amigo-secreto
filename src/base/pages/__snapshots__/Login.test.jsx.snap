// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagina de Login deve renderizar sem problemas 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Grid
        style={
                Object {
                        "height": "100%",
                      }
        }
        textAlign="center"
        verticalAlign="middle"
>
        <GridColumn
                style={
                        Object {
                                "maxWidth": 450,
                              }
                }
        >
                <Header
                        as="h4"
                        textAlign="center"
                >
                        <Header
                                as="h1"
                                color="orange"
                                style={
                                        Object {
                                                "marginBottom": 16,
                                              }
                                }
                        >
                                Amigo Secreto
                        </Header>
                         Acesse sua conta
                </Header>
                <Form
                        as="form"
                        size="large"
                >
                        <Segment
                                stacked={true}
                        >
                                <FormInput
                                        as={[Function]}
                                        control={[Function]}
                                        fluid={true}
                                        icon="user"
                                        iconPosition="left"
                                        id="email"
                                        onChange={[Function]}
                                        placeholder="E-mail"
                                />
                                <FormInput
                                        as={[Function]}
                                        control={[Function]}
                                        fluid={true}
                                        icon="lock"
                                        iconPosition="left"
                                        id="pass"
                                        onChange={[Function]}
                                        placeholder="Senha"
                                        type="password"
                                />
                                <Button
                                        as="button"
                                        color="orange"
                                        fluid={true}
                                        size="large"
                                >
                                        Entrar
                                </Button>
                        </Segment>
                </Form>
                <Message>
                        Novo usuário? 
                        <a
                                href="#"
                        >
                                Cadastre-se
                        </a>
                </Message>
        </GridColumn>
</Grid>,
      "className": "login-form",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <GridColumn
          style={
                    Object {
                              "maxWidth": 450,
                            }
          }
>
          <Header
                    as="h4"
                    textAlign="center"
          >
                    <Header
                              as="h1"
                              color="orange"
                              style={
                                        Object {
                                                  "marginBottom": 16,
                                                }
                              }
                    >
                              Amigo Secreto
                    </Header>
                     Acesse sua conta
          </Header>
          <Form
                    as="form"
                    size="large"
          >
                    <Segment
                              stacked={true}
                    >
                              <FormInput
                                        as={[Function]}
                                        control={[Function]}
                                        fluid={true}
                                        icon="user"
                                        iconPosition="left"
                                        id="email"
                                        onChange={[Function]}
                                        placeholder="E-mail"
                              />
                              <FormInput
                                        as={[Function]}
                                        control={[Function]}
                                        fluid={true}
                                        icon="lock"
                                        iconPosition="left"
                                        id="pass"
                                        onChange={[Function]}
                                        placeholder="Senha"
                                        type="password"
                              />
                              <Button
                                        as="button"
                                        color="orange"
                                        fluid={true}
                                        size="large"
                              >
                                        Entrar
                              </Button>
                    </Segment>
          </Form>
          <Message>
                    Novo usuário? 
                    <a
                              href="#"
                    >
                              Cadastre-se
                    </a>
          </Message>
</GridColumn>,
        "style": Object {
          "height": "100%",
        },
        "textAlign": "center",
        "verticalAlign": "middle",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Header
              as="h4"
              textAlign="center"
>
              <Header
                            as="h1"
                            color="orange"
                            style={
                                          Object {
                                                        "marginBottom": 16,
                                                      }
                            }
              >
                            Amigo Secreto
              </Header>
               Acesse sua conta
</Header>,
            <Form
              as="form"
              size="large"
>
              <Segment
                            stacked={true}
              >
                            <FormInput
                                          as={[Function]}
                                          control={[Function]}
                                          fluid={true}
                                          icon="user"
                                          iconPosition="left"
                                          id="email"
                                          onChange={[Function]}
                                          placeholder="E-mail"
                            />
                            <FormInput
                                          as={[Function]}
                                          control={[Function]}
                                          fluid={true}
                                          icon="lock"
                                          iconPosition="left"
                                          id="pass"
                                          onChange={[Function]}
                                          placeholder="Senha"
                                          type="password"
                            />
                            <Button
                                          as="button"
                                          color="orange"
                                          fluid={true}
                                          size="large"
                            >
                                          Entrar
                            </Button>
              </Segment>
</Form>,
            <Message>
              Novo usuário? 
              <a
                            href="#"
              >
                            Cadastre-se
              </a>
</Message>,
          ],
          "style": Object {
            "maxWidth": 450,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h4",
              "children": Array [
                <Header
                  as="h1"
                  color="orange"
                  style={
                                    Object {
                                                      "marginBottom": 16,
                                                    }
                  }
>
                  Amigo Secreto
</Header>,
                " Acesse sua conta",
              ],
              "textAlign": "center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "h1",
                  "children": "Amigo Secreto",
                  "color": "orange",
                  "style": Object {
                    "marginBottom": 16,
                  },
                },
                "ref": null,
                "rendered": "Amigo Secreto",
                "type": [Function],
              },
              " Acesse sua conta",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "form",
              "children": <Segment
                stacked={true}
>
                <FormInput
                                as={[Function]}
                                control={[Function]}
                                fluid={true}
                                icon="user"
                                iconPosition="left"
                                id="email"
                                onChange={[Function]}
                                placeholder="E-mail"
                />
                <FormInput
                                as={[Function]}
                                control={[Function]}
                                fluid={true}
                                icon="lock"
                                iconPosition="left"
                                id="pass"
                                onChange={[Function]}
                                placeholder="Senha"
                                type="password"
                />
                <Button
                                as="button"
                                color="orange"
                                fluid={true}
                                size="large"
                >
                                Entrar
                </Button>
</Segment>,
              "size": "large",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    fluid={true}
                    icon="user"
                    iconPosition="left"
                    id="email"
                    onChange={[Function]}
                    placeholder="E-mail"
/>,
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    fluid={true}
                    icon="lock"
                    iconPosition="left"
                    id="pass"
                    onChange={[Function]}
                    placeholder="Senha"
                    type="password"
/>,
                  <Button
                    as="button"
                    color="orange"
                    fluid={true}
                    size="large"
>
                    Entrar
</Button>,
                ],
                "stacked": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": [Function],
                    "control": [Function],
                    "fluid": true,
                    "icon": "user",
                    "iconPosition": "left",
                    "id": "email",
                    "onChange": [Function],
                    "placeholder": "E-mail",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": [Function],
                    "control": [Function],
                    "fluid": true,
                    "icon": "lock",
                    "iconPosition": "left",
                    "id": "pass",
                    "onChange": [Function],
                    "placeholder": "Senha",
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "children": "Entrar",
                    "color": "orange",
                    "fluid": true,
                    "size": "large",
                  },
                  "ref": null,
                  "rendered": "Entrar",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Novo usuário? ",
                <a
                  href="#"
>
                  Cadastre-se
</a>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Novo usuário? ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cadastre-se",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Cadastre-se",
                "type": "a",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Grid
          style={
                    Object {
                              "height": "100%",
                            }
          }
          textAlign="center"
          verticalAlign="middle"
>
          <GridColumn
                    style={
                              Object {
                                        "maxWidth": 450,
                                      }
                    }
          >
                    <Header
                              as="h4"
                              textAlign="center"
                    >
                              <Header
                                        as="h1"
                                        color="orange"
                                        style={
                                                  Object {
                                                            "marginBottom": 16,
                                                          }
                                        }
                              >
                                        Amigo Secreto
                              </Header>
                               Acesse sua conta
                    </Header>
                    <Form
                              as="form"
                              size="large"
                    >
                              <Segment
                                        stacked={true}
                              >
                                        <FormInput
                                                  as={[Function]}
                                                  control={[Function]}
                                                  fluid={true}
                                                  icon="user"
                                                  iconPosition="left"
                                                  id="email"
                                                  onChange={[Function]}
                                                  placeholder="E-mail"
                                        />
                                        <FormInput
                                                  as={[Function]}
                                                  control={[Function]}
                                                  fluid={true}
                                                  icon="lock"
                                                  iconPosition="left"
                                                  id="pass"
                                                  onChange={[Function]}
                                                  placeholder="Senha"
                                                  type="password"
                                        />
                                        <Button
                                                  as="button"
                                                  color="orange"
                                                  fluid={true}
                                                  size="large"
                                        >
                                                  Entrar
                                        </Button>
                              </Segment>
                    </Form>
                    <Message>
                              Novo usuário? 
                              <a
                                        href="#"
                              >
                                        Cadastre-se
                              </a>
                    </Message>
          </GridColumn>
</Grid>,
        "className": "login-form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridColumn
            style={
                        Object {
                                    "maxWidth": 450,
                                  }
            }
>
            <Header
                        as="h4"
                        textAlign="center"
            >
                        <Header
                                    as="h1"
                                    color="orange"
                                    style={
                                                Object {
                                                            "marginBottom": 16,
                                                          }
                                    }
                        >
                                    Amigo Secreto
                        </Header>
                         Acesse sua conta
            </Header>
            <Form
                        as="form"
                        size="large"
            >
                        <Segment
                                    stacked={true}
                        >
                                    <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                fluid={true}
                                                icon="user"
                                                iconPosition="left"
                                                id="email"
                                                onChange={[Function]}
                                                placeholder="E-mail"
                                    />
                                    <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                fluid={true}
                                                icon="lock"
                                                iconPosition="left"
                                                id="pass"
                                                onChange={[Function]}
                                                placeholder="Senha"
                                                type="password"
                                    />
                                    <Button
                                                as="button"
                                                color="orange"
                                                fluid={true}
                                                size="large"
                                    >
                                                Entrar
                                    </Button>
                        </Segment>
            </Form>
            <Message>
                        Novo usuário? 
                        <a
                                    href="#"
                        >
                                    Cadastre-se
                        </a>
            </Message>
</GridColumn>,
          "style": Object {
            "height": "100%",
          },
          "textAlign": "center",
          "verticalAlign": "middle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Header
                as="h4"
                textAlign="center"
>
                <Header
                                as="h1"
                                color="orange"
                                style={
                                                Object {
                                                                "marginBottom": 16,
                                                              }
                                }
                >
                                Amigo Secreto
                </Header>
                 Acesse sua conta
</Header>,
              <Form
                as="form"
                size="large"
>
                <Segment
                                stacked={true}
                >
                                <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                fluid={true}
                                                icon="user"
                                                iconPosition="left"
                                                id="email"
                                                onChange={[Function]}
                                                placeholder="E-mail"
                                />
                                <FormInput
                                                as={[Function]}
                                                control={[Function]}
                                                fluid={true}
                                                icon="lock"
                                                iconPosition="left"
                                                id="pass"
                                                onChange={[Function]}
                                                placeholder="Senha"
                                                type="password"
                                />
                                <Button
                                                as="button"
                                                color="orange"
                                                fluid={true}
                                                size="large"
                                >
                                                Entrar
                                </Button>
                </Segment>
</Form>,
              <Message>
                Novo usuário? 
                <a
                                href="#"
                >
                                Cadastre-se
                </a>
</Message>,
            ],
            "style": Object {
              "maxWidth": 450,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h4",
                "children": Array [
                  <Header
                    as="h1"
                    color="orange"
                    style={
                                        Object {
                                                            "marginBottom": 16,
                                                          }
                    }
>
                    Amigo Secreto
</Header>,
                  " Acesse sua conta",
                ],
                "textAlign": "center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "h1",
                    "children": "Amigo Secreto",
                    "color": "orange",
                    "style": Object {
                      "marginBottom": 16,
                    },
                  },
                  "ref": null,
                  "rendered": "Amigo Secreto",
                  "type": [Function],
                },
                " Acesse sua conta",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "form",
                "children": <Segment
                  stacked={true}
>
                  <FormInput
                                    as={[Function]}
                                    control={[Function]}
                                    fluid={true}
                                    icon="user"
                                    iconPosition="left"
                                    id="email"
                                    onChange={[Function]}
                                    placeholder="E-mail"
                  />
                  <FormInput
                                    as={[Function]}
                                    control={[Function]}
                                    fluid={true}
                                    icon="lock"
                                    iconPosition="left"
                                    id="pass"
                                    onChange={[Function]}
                                    placeholder="Senha"
                                    type="password"
                  />
                  <Button
                                    as="button"
                                    color="orange"
                                    fluid={true}
                                    size="large"
                  >
                                    Entrar
                  </Button>
</Segment>,
                "size": "large",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <FormInput
                      as={[Function]}
                      control={[Function]}
                      fluid={true}
                      icon="user"
                      iconPosition="left"
                      id="email"
                      onChange={[Function]}
                      placeholder="E-mail"
/>,
                    <FormInput
                      as={[Function]}
                      control={[Function]}
                      fluid={true}
                      icon="lock"
                      iconPosition="left"
                      id="pass"
                      onChange={[Function]}
                      placeholder="Senha"
                      type="password"
/>,
                    <Button
                      as="button"
                      color="orange"
                      fluid={true}
                      size="large"
>
                      Entrar
</Button>,
                  ],
                  "stacked": true,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "control": [Function],
                      "fluid": true,
                      "icon": "user",
                      "iconPosition": "left",
                      "id": "email",
                      "onChange": [Function],
                      "placeholder": "E-mail",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": [Function],
                      "control": [Function],
                      "fluid": true,
                      "icon": "lock",
                      "iconPosition": "left",
                      "id": "pass",
                      "onChange": [Function],
                      "placeholder": "Senha",
                      "type": "password",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "button",
                      "children": "Entrar",
                      "color": "orange",
                      "fluid": true,
                      "size": "large",
                    },
                    "ref": null,
                    "rendered": "Entrar",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Novo usuário? ",
                  <a
                    href="#"
>
                    Cadastre-se
</a>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Novo usuário? ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cadastre-se",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": "Cadastre-se",
                  "type": "a",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
